sympy/concrete/tests/test_sums_products.py

```diff
@@ -886,11 +886,7 @@ def test_issue_2787():
     binomial_dist = binomial(n, k)*p**k*(1 - p)**(n - k)
     s = Sum(binomial_dist*k, (k, 0, n))
     res = s.doit().simplify()
-    assert res == Piecewise(
-        (n*p, p/Abs(p - 1) <= 1),
-        ((-p + 1)**n*Sum(k*p**k*(-p + 1)**(-k)*binomial(n, k), (k, 0, n)),
-        True))
-
+    assert res == n*p
```

Comment:
Hey,

Sorry I haven't looked at this in quite some time now. Nonetheless I can interpret this bit, 
It deals with the sum of binomial(n, k)*p**k*(1 - p)**(n - k) * k from k = 0 to n

The previous test asserts it to be np when p/|p-1| <= 1 and otherwise it was just returning the formal sum, this result is certainly correct but in fact this business of p/|p-1| <= 1 is needless, the sum is true for all real p. Presumably however the sum was being done prior it was doing something that just proves things on some restricted domain.

I think this is probably what a discussion between me and asmeurer a number of posts up concerned but I don't really remember.